### ISSUE TITLE
Mention the @trace macro in readme

### ISSUE THREAD

**Issue by Evizero** [JuliaML/ValueHistories.jl#9]
We should advertise this functionality

**Comment by tbreloff** [comment]
Yes, though I'm not sure its ready for prime time yet. I want to use it a
bit and maybe make some changes. I was thinking about a second "trace_iter"
macro which does what trace currently does. It would also be nice to also
support "dynamic tables" with strongly typed fields under the same
framework for those cases where you always add the same fields in a loop
and want performance. We should examine the DataStreams world and see if we
can overlap.

**Comment by Evizero** [comment]
all good ideas

**Comment by cossio** [comment]
Bump?

**Comment by cossio** [comment]
It would be nice that the `@trace` macro had a variant where you didn't have to specify the iteration number, so it would just `push!` and increment by one whatever the previous iteration number was.
